docs/rules-engine/how-to-use/build-in-facts.md

```diff
@@ -0,0 +1,155 @@
+# Rules engine - Build-in facts
+
+In some cases, the Otter [operators](../README.md#operator) may refer to an implicit fact. This is the case for
+the ``dateInNextMinutes`` and ``dateNotInNextMinutes`` operators.
+
+The definition of these two operators implies the notion of a current date, hence, the framework needs to expose an 
+internal, build-in fact that will always be used by them: the ``o3rCurrentTime``. The operators will not take it as input, 
+making it an implicit fact of these operators. This also means that, whereas all the other operators are pure function,
+this is not the case for ``dateInNextMinutes`` nor ``dateNotInNextMinutes``.
+
+While this fact is provided in the rules engine module, you should be able to override its implementation to fit your needs.
+
+This section explains how to use operators with an implicit fact and how to override the fact implementation.
+
+## Usage
+The operators using an implicit, build-in fact is used in the same way as any other operator, the implicit fact will just
+not be referred to in the Ruleset object:
+```json55
+{
+  "id": "5467e501-b9ff-414f-8026-56885d0d7a4c",
+  "name": "The otter is late",
+  "disabled": false,
+  "outputRuntimeFacts": [],
+  "inputRuntimeFacts": [],
+  "rootElement": {
+    "elementType": "RULE_BLOCK",
+    "blockType": "IF_ELSE",
+    "condition": {
+      "all": [
+        {
+          "lhs": {
+            "type": "FACT",
+            "value": "outboundDate"
+          },
+          "rhs": {
+              "type": "LITERAL",
+              "value": "2880"
+          },
+          "operator": "dateInNextMinutes"
+        }
+      ]
+    },
+    "successElements": [
+      {
+        "elementType": "ACTION",
+        "actionType": "UPDATE_ASSET",
+        "asset": "otter.svg",
+        "value": "otter-late.svg"
+      }
+    ],
+    "failureElements": []
+  }
+}
+```
+
+This does not mean that the fact itself should not be registered.
+If you fail to register the fact the same way as the non-build-in facts, the whole ruleset will fail.
+
+```typescript
+import {Component, OnInit} from '@angular/core';
+import {CurrentTimeFactsService} from '@o3r/rules-engine';
```

Comment:
This service does not come from '@o3r/rules-engine' but should be implemented on each application

Reply:
Indeed, I was sure we did provide it

Reply:
As discussed, I will keep this documentation but we will expose the CurrentTimeFactsService in the rules-engine package